Add test to check and log mdatp install/config info

mcgov opened the pull request:
Adding an mdatp test to check configuration info in a VM if defender is installed.

mcgov committed:
MDATP: add mdatp existence and config test

- stop lisa from overwriting lf to crlf in bash files

- mdatp script to check for json files, ascii or utf8 text

script fails on detection of any logs or installation,
uses exit code as issue signal instead of string parsing

srkoripe reviewed (approved):
LGTM from mdatp side. Pls get a review from the MDC agent side too if possible

mcgov committed:
shorten line, switch to !windows as the condition

mcgov committed:
mdatp: code review, Fix 'found' in comments

mcgov committed:
mdatp: code review part six

- Add more comments describing the code paths
- Add explanation of SKIP if an unexpected error code is found.
	this implies we're not running in a unix shell,
	which should not occur, but anything is possible.
	If that occurs, skip instead of fail or pass.

mcgov committed:
mdatp: swap from bitmask to byte value with precedence

-Move to using an int as return code to avoid conflicting
with existing shell exit codes.

This is done to help differentiate between os/shell errors and
a successful validation (either fail or pass)

mcgov committed:
Make error messages more informative and actionable, change 'unvalidated' code path to a failure with prompt to open a bug
